File: core_common/src/main/java/com/huyingbao/core/common/model/CommonHttpException.java
package com.huyingbao.core.common.model;


import java.io.IOException;

import androidx.annotation.Nullable;

/**
 * 自定义HttpException
 *
 * @author liujunfeng
 * @date 2019/1/1
 */
public class CommonHttpException extends IOException {
    private final int mCode;

    private final String mMessage;

    public CommonHttpException(int code, @Nullable String message) {
        super("HTTP " + code + " " + message);
        this.mCode = code;
        this.mMessage = message;
    }

    /**
     * HTTP status mCode.
     */
    public int code() {
        return mCode;
    }

    /**
     * HTTP status mMessage.
     */
    public String message() {
        return mMessage;
    }
}

File: core_common_kotlin/src/main/java/com/huyingbao/core/common/model/CommonHttpException.kt
package com.huyingbao.core.common.model


import java.io.IOException

/**
 * 自定义HttpException
 * Created by liujunfeng on 2019/1/1.
 */
class CommonHttpException(private val mCode: Int, private val mMessage: String) : IOException("HTTP $mCode $mMessage") {

    /**
     * HTTP status mCode.
     */
    fun code(): Int {
        return mCode
    }

    /**
     * HTTP status mMessage.
     */
    fun message(): String {
        return mMessage
    }
}
